rz-cmn-srp: rebase the latest changes from the styhead/rz-sbc branch

tiennguyenuh opened the pull request:
Rebase the latest changes from the styhead/rz-sbc branch

tiennguyenuh committed:
rz-sbc: update readme files

Update readme to clarify the purpose of each folder.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add docker support option

Changes include:
	- Add docker support option in `local.conf.sample`
	- Update docker.cfg

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: Update renesas-ubuntu.bb

Changes include:
	- Clean duplicate configuration in `renesas-ubuntu.bb` and fix
	  the issue of creating esdk that needs `environment-setup-append.sh`.
	- Add clean up call in `SDK_POSTPROCESS_COMMAND` after building SDK.
	- Move the renesas-ubuntu recipe into each qt version folder.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
gstreamer-omx: change submodule url

Using a patch to update the submodule URL for gst-omx from
https://gitlab.freedesktop.org/gstreamer/common.git to
https://github.com/GStreamer/common.git

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: correct addtask ordering in bsp recipes

Correct addtask ordering in firmware-package recipes

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add guide to switch SSH server and add GPG key

Changes include:
	- Add guide to switch from OpenSSH to Dropbear.
	- Install `trusted.gpg` key by default for APT repository.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on dynamic-layers/qt5-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+SUMMARY = "Custom RZ/G2L-SBC Image"
```
Change the name

vudangRVC commented on dynamic-layers/qt5-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+
+# compatible machine comes with linux-yocto but not available in this build
+# so bring back these parameters
+COMPATIBLE_MACHINE = "(rzg2l-sbc)"
```
The machine is wrong, ubuntu-image supports on all machines

vudangRVC commented on dynamic-layers/qt5-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+IMAGE_FSTYPES = " tar.bz2"
+# bootloader for rzsbc
+DEPENDS += " firmware-pack"
+MACHINEOVERRIDES =. "rzg2l:"
```
Update the machine

vudangRVC commented on dynamic-layers/qt6-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+SUMMARY = "Custom RZ/G2L-SBC Image"
```
Change the name

vudangRVC commented on dynamic-layers/qt6-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+
+# compatible machine comes with linux-yocto but not available in this build
+# so bring back these parameters
+COMPATIBLE_MACHINE = "(rzg2l-sbc)"
```
The machine is wrong, ubuntu-image supports on all machines

vudangRVC commented on dynamic-layers/qt6-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+IMAGE_FSTYPES = " tar.bz2"
+# bootloader for rzsbc
+DEPENDS += " firmware-pack"
+MACHINEOVERRIDES =. "rzg2l:"
```
Update the machine

vudangRVC commented on dynamic-layers/qt5-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+SUMMARY = "Custom RZ/G2L-SBC Image"
+LICENSE = "MIT"
+require include/core-image-renesas-mmp.inc
+require include/core-image-renesas-qt.inc
```
No longer support

vudangRVC commented on dynamic-layers/qt6-layer/images/renesas-ubuntu.bb:
```diff
@@ -0,0 +1,44 @@
+SUMMARY = "Custom RZ/G2L-SBC Image"
+LICENSE = "MIT"
+require include/core-image-renesas-mmp.inc
+require include/core-image-renesas-qt.inc
```
No longer support

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
```diff
@@ -89,134 +91,267 @@ $ IMAGE=<target_image> ./rzsbc_yocto.sh build
+│   │           │       ├── set_swap_enable.sh
+│   │           │       └── setup-set-permissions.sh
+│   │           └── setup_ubuntu_environment.sh
+│   └── tools
```
Please update this folder after universal scripts integrated

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
```diff
@@ -89,134 +91,267 @@ $ IMAGE=<target_image> ./rzsbc_yocto.sh build
-    │   └── rootfs                                                       <---- Contains compressed root filesystem images
-    │       ├── renesas-core-image-weston-rzg2l-sbc.tar.bz2
-    │       └── Readme.md
+└── target
```
The same. Make sure you update this after the integration build complete

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
````diff
@@ -226,46 +361,58 @@ The eSDK build generates an installer, which you will use to install the eSDK on
+$ source ~/esdk/5.1.4/environment-setup-armv8-2a-poky-linux
 ```
 
 ## Programming/Flashing images for RZG2L-SBC
````
Update this section

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
````diff
@@ -333,31 +484,27 @@ In U-Boot console, execute one more command to bring RZG2L-SBC system up:
 => boot
 ```
-
 ## Confirm supported features on RZG2L-SBC
````
Update this section. Make sure each board has each sub-section.

vudangRVC commented on recipes-extend/universal-scripts/universal-scripts_1.0.bb:
```diff
@@ -0,0 +1,20 @@
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+URL = "git://github.com/vudangRVC/rz-utility.git"
```
Change this URL

vudangRVC replied:
Repo: https://github.com/Renesas-SST/rz-utils
Branch: rz-utils-init

vudangRVC commented on recipes-extend/target-readme/files/Readme.md:
We still need this file, don't we?

Also for all the files in this target-readme folder.

tiennguyenuh replied:
No, we don't need them, the file was already deleted.

vudangRVC commented on recipes-extend/host-readme/files/Readme.md:
We still need this file, right? 

And all the readme files in this host-readme folder

tiennguyenuh committed:
rz-cmn: Update README and rename Qt framework include file

Changes include:
	- Update the main README to reflect compatibility with new machine targets
	- Rename Qt include file for clarity and consistency with its Qt-specific usage

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh commented:
Hi @vudangRVC ,

I've addressed all feedback so far in this commit: https://github.com/Renesas-SST/meta-renesas/commit/b9bb71b928ec09a6b6a989c2ea2fc5cdee7a0304. Please help me to check if everything looks good to you.

tiennguyenuh committed:
rz-cmn: add universal-scripts task

Changes include:
	- Add universal-scripts task to create a host tools directory
	  support for rz-cmn
	- Remove un-use recipes

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on recipes-extend/host-readme/files/Readme.md:
```diff
@@ -1,38 +1,35 @@
-# RZ Common host folder
+# HHost folder
```
Typo

tiennguyenuh replied:
Updated.

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
```diff
@@ -1,954 +1,539 @@
+## 1. Overview
 
-Known issues:
+This release is based on the latest VLP of the Renesas RZ/V2L and RZ/G2L development product. It provides a comprehensive Linux BSP (Board Support Package) with various features and tools for developing embedded applications on the supported Renesas boards.
```
We also support V2H as well. Please update this paragraph

tiennguyenuh replied:
Updated.

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
```diff
@@ -1845,10 +1444,34 @@ Step 1: Configure Eclipse to connect to the GDB Server:
 - Utilize the Expressions view to evaluate any expressions or check the state of specific variables.
 - Navigate through the call stack to see the sequence of function calls leading to the crash. This can provide insight into how the program reached the faulting line.
+
+### 4.2. BSP Interface for RZ/G2L-SBC and RZ/G2L, RZ/V2L Evaluation Boards
```
Seperate the contents of RZ/G2L-SBC vs RZG2L-EVK and RZV2L-EVK. We've already had a full section for RZG2L-SBC.

For RZG2L-EVK and RZV2L-EVK, the same content as RZV2H-EVK is enough

tiennguyenuh replied:
Updated.

vudangRVC commented on recipes-docs/rz-cmn-readme/files/README.md:
````diff
@@ -1369,28 +968,28 @@ $ echo ${GDB}
 ```
 
-#### Install GDBServer on RZG2L-SBC
+##### 4.1.5.2. Install GDBServer on RZG2L-SBC
````
Only on RZG2L-SBC?

tiennguyenuh replied:
It is a common features section, so it also applies to other supported boards. I've updated the README accordingly.
